pyrefly/lib/alt/expr.rs

```diff
@@ -187,7 +189,14 @@ impl<'a, Ans: LookupAnswer> AnswersSolver<'a, Ans> {
     fn ifs_infer(&self, comps: &[Comprehension], errors: &ErrorCollector) {
         for comp in comps {
             for if_clause in comp.ifs.iter() {
-                self.expr_infer(if_clause, errors);
+                let if_wrapper = Expr::If(ExprIf {
+                    test: Box::new(if_clause.clone()),
+                    body: Box::new(Expr::NoneLiteral(Default::default())),
+                    orelse: Box::new(Expr::NoneLiteral(Default::default())),
+                    range: if_clause.range(),
```

Comment:
I think we could avoid this wrapper here by restructuring the code a little bit. 

I think we could have a separate helper method which takes a type and performs the truthy check on it. Then we could simply pass the return of `self.expr_infer(if_clause, errors)` to that method here, as opposed to passing the wrapper to `expr_infer()`. 

The same helper method could then be reused below in `expr_infer_type_no_trace()` and `solve_expectations()`.